RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorKernelsImpl.h

```diff
@@ -561,11 +561,12 @@ __global__ void kernel_fillHitDetIndices(HitContainer const *__restrict__ tuples
 
 __global__ void kernel_fillNLayers(TkSoA *__restrict__ ptracks, cms::cuda::AtomicPairCounter *apc) {
   auto &tracks = *ptracks;
-  auto first = blockIdx.x * blockDim.x + threadIdx.x;
-  auto ntracks = apc->get().m;
+  uint32_t first = blockIdx.x * blockDim.x + threadIdx.x;
+  // clamp the number of tracks to the capacity of the SoA
+  auto ntracks = std::min<uint32_t>(apc->get().m, tracks.stride() - 1);
   if (0 == first)
     tracks.setNTracks(ntracks);
-  for (int idx = first, nt = ntracks; idx < nt; idx += gridDim.x * blockDim.x) {
+  for (uint32_t idx = first; idx < ntracks; idx += gridDim.x * blockDim.x) {
     auto nHits = tracks.nHits(idx);
```

Comment:
why?
int was intentional.

Reply:
which change ?
  - `nt = ntracks`: I don't see how it can make a difference (`ntracks` is also a local variable) but I can revert it
  - `int` vs `uint32_t`: becuase `ntracks` is now `uint32_t` and the compiler warns about signed-vs-unsigned comparison